src/roswire/ros2/state.py

```diff
@@ -14,12 +18,55 @@
 class ROS2StateProbe:
     """Provides an interface for obtaining the instantaneous state of a ROS
     system in terms of its publishers, subscribers, and services."""
+    _app_instance: 'AppInstance' = attr.ib()
+
     @classmethod
     def for_app_instance(cls, app_instance: 'AppInstance') -> 'ROS2StateProbe':
-        raise NotImplementedError
+        return ROS2StateProbe(app_instance=app_instance)
 
     def probe(self) -> SystemState:
         """Obtains the instantaneous state of the associated ROS system."""
-        raise NotImplementedError
+        node_to_state: Dict[Optional[str], Dict[str, List[str]]] = {'pub': {},
+                                                                    'sub': {},
+                                                                    'serv': {}}
+        command = "rosnode list"
+        try:
+            output = self._app_instance.shell.check_output(command, text=True)
+        except dockerblade.exceptions.CalledProcessError as error:
+            logger.debug("Unable to retrieve rosnode list from command line")
+            raise error
+        node_names = output.split('\r\n')
+        for node_name in node_names:
+            info = f"ros2 node info '{node_name}'"
```

Comment:
Let's rename this to `command_info` to make it clear that this is a command (rather than the output from the command).